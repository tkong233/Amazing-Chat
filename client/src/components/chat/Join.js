import React from 'react';

class Join extends React.Component {
  render() {
    const { onSubmit, onChange, username, room } = this.props;
    return (
      <div className="container" data-test = "JoinComponent">
        <div className="row">
        <div className="col s8 offset-s2">
        <form noValidate onSubmit={onSubmit}>
          {/* Name */}
          <div className="input-field col s12">
            <input
            onChange={onChange}
            value={username}
            id="username"
            type="text"
          />
            <label htmlFor="username">Username</label>
          </div>

          {/* Room */}
          <div className="input-field col s12">
            <input
            onChange={onChange}
            value={room}
            id="room"
            type="text"
          />
            <label htmlFor="room">Room</label>
          </div>

          {/* Submit Button */}
          <div className="col s12" style={{ paddingLeft: "11.250px" }}>
            <button
            style={{
                width: "150px",
                borderRadius: "3px",
                letterSpacing: "1.5px",
                marginTop: "1rem"
            }}
            type="submit"
            className="btn waves-effect waves-light hoverable blue-grey darken-2"
            >
            Join
            </button>
          </div>
        </form>
        </div>
        </div>
        </div>
    )
  }
}

export default Join;