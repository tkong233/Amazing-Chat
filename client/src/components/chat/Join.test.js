import React from 'react';
import Join from './Join';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16'; 
import checkPropTypes from 'check-prop-types';

Enzyme.configure({
    adapter: new EnzymeAdapter(),
});


describe('Message Component', ()=>{
    describe('Checking PropTypes', ()=>{
        it('Should not throw a warning', ()=>{
            const expectedProps = {
                onSubmit: ()=>{},
                onChange: ()=>{},
                username: '',
                room: ''
            }
            // eslint-disable-next-line react/forbid-foreign-prop-types
            const propsErr = checkPropTypes(Join.propTypes, expectedProps, 'props', Join.name);
            expect(propsErr).toBeUndefined();
        })
    });
    describe('Renders', ()=>{
        let wrapper;
        beforeEach(()=>{
            const props = {
                onSubmit: ()=>{},
                onChange: ()=>{},
                username: '',
                room: ''
            };
            wrapper = shallow(<Join {...props} />);
        });
        
        it('Should render join', ()=>{
            const join = wrapper.find(`[data-test='JoinComponent']`);
            expect(join.length).toBe(1);
        })
        

    });
});
