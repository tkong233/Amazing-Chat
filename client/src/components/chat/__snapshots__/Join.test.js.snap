// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join Component Renders Snapshot testing 1`] = `ShallowWrapper {}`;
